Test logic doesn't match PDM's actual behavior

### Describe the bug

Hello guys

After running the tests i got the `test test_use_python_write_file_multiple_versions`failing. 

The test creates a `.python-version` file with unavailable Python versions (in my case it was only 3.10) and expects PDM to ignore them, but PDM's actual behavior is to auto-install missing Python versions when they're requested in single-line `.python-version` files, or ignore multiline files entirely. 

The test is failing because PDM correctly auto-installs Python 3.10 after finding it in the .python-version file, while the test expects PDM to use a different available version instead. 

### To reproduce

- Make sure you have only one of the `DEFAULT_PYTHON_VERSIONS` missing, in my case `3.10`
Run 
- run `pdm run pytest tests/test_integration.py::test_use_python_write_file_multiple_versions -v`
- You are going to get `AssertionError: assert '3.10' not in ['3.10']` 

### Expected Behavior

The test `test_use_python_write_file_multiple_versions` and do the following: when a .python-version file contains a single Python version (available or unavailable), PDM should use that version and auto-install it if necessary, and when the file contains multiple versions (making it multiline), PDM should ignore the file entirely and fall back to using an available Python version from the system

### Environment Information

{
  "implementation_name": "cpython",
  "implementation_version": "3.11.9",
  "os_name": "posix",
  "platform_machine": "arm64",
  "platform_release": "25.0.0",
  "platform_system": "Darwin",
  "platform_version": "Darwin Kernel Version 25.0.0: Wed Sep 17 21:42:08 PDT 2025; root:xnu-12377.1.9~141/RELEASE_ARM64_T8132",
  "python_full_version": "3.11.9",
  "platform_python_implementation": "CPython",
  "python_version": "3.11",
  "sys_platform": "darwin"
}

### Verbose Command Output

_No response_

### Additional Context

_No response_

### Are you willing to submit a PR to fix this bug?

- [x] Yes, I would like to submit a PR.

--- a/news/3660.bugfix.md
+++ b/news/3660.bugfix.md
@@ -1,0 +1,1 @@
+Fix test_use_python_write_file_multiple_versions to match PDM's actual behavior.

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -91,11 +91,20 @@ def test_init_project_respect_version_file(pdm, project, python_version, via_env
 def test_use_python_write_file_multiple_versions(pdm, project, python_version, monkeypatch):
     no_versions = [p for p in DEFAULT_PYTHON_VERSIONS if p not in PYTHON_VERSIONS]
     project.project_config["python.use_venv"] = True
-    project.root.joinpath(".python-version").write_text("\n".join(no_versions))
+
+    if no_versions and PYTHON_VERSIONS:
+        version_content = f"{no_versions[0]}\n{PYTHON_VERSIONS[0]}"
+    elif no_versions:
+        version_content = "\n".join(no_versions)
+    else:
+        version_content = "\n".join(PYTHON_VERSIONS[:2])
+
+    project.root.joinpath(".python-version").write_text(version_content)
     project._saved_python = None
     project._environment = None
     pdm(["install"], obj=project, strict=True)
-    assert f"{project.python.major}.{project.python.minor}" not in no_versions
+
+    assert f"{project.python.major}.{project.python.minor}" in PYTHON_VERSIONS
 
 
 @pytest.mark.integration

